pkg/cmd/cli/cli.go

```diff
@@ -62,43 +62,76 @@ func NewCommandCLI(name, fullName string) *cobra.Command {
-	cmds.AddCommand(cmd.NewCmdRollback(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdEnv(fullName, f, os.Stdin, out))
-	cmds.AddCommand(cmd.NewCmdExpose(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdGet(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDescribe(fullName, f, out))
-	// Deprecate 'oc apply' with 'oc create' command.
-	cmds.AddCommand(applyToCreate(cmd.NewCmdCreate(fullName, f, out)))
-	cmds.AddCommand(cmd.NewCmdProcess(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdEdit(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdUpdate(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdDelete(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdLogs(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdImportImage(fullName, f, out))
-	cmds.AddCommand(cmd.NewCmdExec(fullName, f, os.Stdin, out, os.Stderr))
-	cmds.AddCommand(cmd.NewCmdPortForward(fullName, f))
-	cmds.AddCommand(cmd.NewCmdProxy(fullName, f, out))
+	groups := templates.CommandGroups{
+		{
+			Message: "Basic Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdLogin(fullName, f, in, out),
+				cmd.NewCmdRequestProject("new-project", fullName+" new-project", fullName+" login", fullName+" project", f, out),
+				cmd.NewCmdNewApplication(fullName, f, out),
+				cmd.NewCmdStatus(fullName, f, out),
+				cmd.NewCmdProject(fullName+" project", f, out),
+			},
+		},
+		{
+			Message: "Build and Deploy Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdStartBuild(fullName, f, out),
+				cmd.NewCmdBuildLogs(fullName, f, out),
+				cmd.NewCmdDeploy(fullName, f, out),
+				cmd.NewCmdRollback(fullName, f, out),
+				cmd.NewCmdNewBuild(fullName, f, out),
+				cmd.NewCmdCancelBuild(fullName, f, out),
+				cmd.NewCmdImportImage(fullName, f, out),
+				cmd.NewCmdScale(fullName, f, out),
+			},
+		},
+		{
+			Message: "Application Modification Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdGet(fullName, f, out),
+				cmd.NewCmdDescribe(fullName, f, out),
+				cmd.NewCmdEdit(fullName, f, out),
+				cmd.NewCmdEnv(fullName, f, os.Stdin, out),
+				cmd.NewCmdLabel(fullName, f, out),
+				cmd.NewCmdExpose(fullName, f, out),
+				cmd.NewCmdStop(fullName, f, out),
+				cmd.NewCmdDelete(fullName, f, out),
+			},
+		},
+		{
+			Message: "Troubleshooting and Debugging Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdLogs(fullName, f, out),
+				cmd.NewCmdExec(fullName, f, os.Stdin, out, os.Stderr),
+				cmd.NewCmdPortForward(fullName, f),
+				cmd.NewCmdProxy(fullName, f, out),
+			},
+		},
+		{
+			Message: "Advanced Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdCreate(fullName, f, out),
+				cmd.NewCmdUpdate(fullName, f, out),
+				cmd.NewCmdProcess(fullName, f, out),
+				policy.NewCommandPolicy(policy.PolicyRecommendedName, fullName+" "+policy.PolicyRecommendedName, f, out),
+			},
+		},
+		{
+			Message: "Settings Commands:",
+			Commands: []*cobra.Command{
+				cmd.NewCmdLogout("logout", fullName+" logout", fullName+" login", f, in, out),
```

Comment:
It's not as common an action - these are grouped to be recognizable to a new user

> On Jun 6, 2015, at 4:19 PM, Michail Kargakis <notifications@github.com> wrote:
> 
> In pkg/cmd/cli/cli.go:
> 
> > +				cmd.NewCmdProxy(fullName, f, out),
> > +			},
> > +		},
> > +		{
> > +			Message: "Advanced Commands:",
> > +			Commands: []*cobra.Command{
> > +				cmd.NewCmdCreate(fullName, f, out),
> > +				cmd.NewCmdUpdate(fullName, f, out),
> > +				cmd.NewCmdProcess(fullName, f, out),
> > +				policy.NewCommandPolicy(policy.PolicyRecommendedName, fullName+" "+policy.PolicyRecommendedName, f, out),
> > +			},
> > +		},
> > +		{
> > +			Message: "Settings Commands:",
> > +			Commands: []*cobra.Command{
> > +				cmd.NewCmdLogout("logout", fullName+" logout", fullName+" login", f, in, out),
> Shouldn't logout be alongside login?
> 
> —
> Reply to this email directly or view it on GitHub.
>